.github/workflows/hdmf_compatibility_schema.yml

```diff
@@ -0,0 +1,27 @@
+name: Check HDMF Dev Compatibility
+
+on: [pull_request, workflow_dispatch]
+
+jobs:
+  check_compatibility:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel non-latest runs
+        uses: styfle/cancel-workflow-action@0.11.0
+        with:
+          all_but_latest: true
+          access_token: ${{ github.token }}
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }} # checkout branch
```

Comment:
I believe this checks out the current branch of this repo in the base directory. But when you run git clone hdmf, you get the version of hdmf-common that is "saved" in the default branch of hdmf. That version will be used when you run pytest, not the version checked out at the base directory which hdmf doesn't know about.

Reply:
I don't want to hardcode the branch name and then have to change it each time I run it. I do see what you mean about the order of operations. I want to see if there's a way to not have manually change the branch name each time I cd into the schema repo. If not, then is this still something we want? @rly

Reply:
In theory, maybe I can pass the branch name in as an expression in the workflow. I'll try that

Reply:
I think `{{ github.ref }}` is the branch name. Not sure how it performs on the main branch but I think it's the same...

Reply:
Yeah that's what I found as well. I'll do that tonight!